packages/api/src/hooks/useDocumentUpload.ts

```diff
@@ -56,12 +57,43 @@ const useDocumentUpload = () => {
                 chunks.forEach(chunk => {
                     const size = numToUint8Array(chunk.length);
                     const payload = new Uint8Array([...type, ...id, ...size, ...chunk]);
-                    activeWebSocket?.send(payload);
+                    derivAPI.connection.send(payload);
                 });
+
+                const timeout = setTimeout(() => {
+                    derivAPI.connection?.removeEventListener('message', handleUploadStatus);
+                }, 20000);
+
+                const handleUploadStatus = (messageEvent: MessageEvent) => {
+                    const data = JSON.parse(messageEvent.data);
+
+                    if (data.error) {
+                        derivAPI.connection?.removeEventListener('message', handleUploadStatus);
+                        setDocumentUploadStatus('error');
+                        clearTimeout(timeout);
+                        return;
+                    }
+
+                    if (data.document_upload && data.document_upload?.status === 'failure') {
+                        derivAPI.connection?.removeEventListener('message', handleUploadStatus);
+                        setDocumentUploadStatus('failure');
+                        clearTimeout(timeout);
+                        return;
+                    }
+
+                    if (data.document_upload && data.document_upload?.status === 'success') {
+                        derivAPI.connection?.removeEventListener('message', handleUploadStatus);
+                        setDocumentUploadStatus('success');
+                        clearTimeout(timeout);
+                    }
```

Comment:
I see we are calling clearTimeout on every if statement, probably we can move out from the if statement and call it just once underneath 🙏

Reply:
Alright!